pkg/endpoint/endpoint.go

```diff
@@ -356,14 +362,107 @@ func (e *Endpoint) WaitForProxyCompletions() error {
 	return nil
 }
 
+// runK8sCiliumEndpointSync starts a controller that syncronises the endpoint
+// to the corresponding k8s CiliumEndpoint CRD
+// ep.Mutex must be help for the setup portion of this function, but not the
+// controller.
+func (e *Endpoint) runK8sCiliumEndpointSync() {
+	elog := e.getLogger()
+	endpointID := e.ID
+
+	controllerName := fmt.Sprintf("sync-to-k8s-ciliumendpoint (%v)", endpointID)
+	epName := fmt.Sprintf("%v-cep-%v", node.GetName(), endpointID)
+
+	if !k8s.IsEnabled() {
+		elog.WithField("name", controllerName).Debug("Not starting controller because k8s is disabled")
+		return
+	}
+
+	restConfig, err := k8s.CreateConfig()
```

Comment:
Isn't there a client in `k8s`? `k8s.Client()`

Reply:
I tried using this but I couldn't figure out how to get from the return of k8s.GetClient() to a cilium object :( I have a getCiliumClient function now, so I can try something out if you can see the path.